patches/tModLoader/Terraria/ModLoader/Core/ModOrganizer.cs

```diff
@@ -126,6 +126,36 @@ private enum SearchFolders { }
 			return true;
 		}
 
+		internal static IEnumerable<ulong> DownloadWorkshopDependencies() {
+			HashSet<ulong> dependencies = new HashSet<ulong>();
+
+			foreach (LocalMod mod in FindWorkshopMods()) {
+				// This shouldn't really ever fail, but better safe than sorry.
+				if (!TryReadManifest(GetParentDir(mod.modFile.path), out var manifest))
```

Comment:
We should consider skip dependencies checks if localMod.modreferences contains no dependencies. 

One less query to make, thus better performance.
Not required, but would be nice

Reply:
👍